components/tikv_util/src/stream.rs

```diff
@@ -123,3 +124,18 @@ where
 
     result
 }
+
+impl<E> RetryError for RusotoError<E> {
```

Comment:
I don't know if this is feasible but can we move this into the `cloud` crate? Basically I want to hide `rusoto` inside the `cloud` crate and other crates in tikv only depend on it independently via the `cloud` crate.

Reply:
rusoto is hidden in the aws crate in my other WIP PR. But rusoto_core may be imported anywhere where a trait is defined due to Rust's restrictions on where traits can be implemented. This is just what we have to do due to the way Rust works.
The only other option is to implement the trait for a new wrapper trait of the rusoto_core type (RusotoError) in this case. I refer to this wrapping as the newtype pattern. I think we could use this pattern in this case, but in general it is not always possible to use it for this purpose. I did already use this pattern once in my WIP to define Debug for a rusoto type when improving encryption.

Reply:
Can we move this `RetryError` impl into cloud crate in your other PR? Let's not worry about other use case of `rusoto` in the rest of non-cloud tikv code.